### PR TITLE
Add google.api.core.helpers.grpc_helpers

### PR DESCRIPTION
**Pull request by jonparrott**

### CODE REVIEW

**Comment by jonparrott** [comment]
Towards #3842

**Comment by jonparrott** [comment]
@dhermes any comments?

**Comment by jonparrott** [comment]
Also, a note on this implementation:

It is intentionally simplistic for now. I've made it possible so that in the future we could return subclasses of the gRPC ABCs (`grpc.UnaryUnaryMultiCallable`, etc.) to handle wrapping errors for *all* of the ways of invoking the rpc (`with_call` and `future` specifically). I don't think this is required for now, but I wanted to make sure we could do it in the future without breaking anything.

File: core/google/api/core/helpers/grpc_helpers.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+
+# The list of gRPC Callable interfaces that return iterators.
+_STREAM_WRAP_CLASSES = (
+    grpc.UnaryStreamMultiCallable,
+    grpc.StreamStreamMultiCallable,
+)
+
+
+def _patch_callable_name(callable):
```
Comment (dhermes) [inline core/google/api/core/helpers/grpc_helpers.py]: `callable` is in `__builtins__` (see how the code highlighting makes it blue below), please don't shadow it. Suggest `callable_`

Comment (jonparrott) [reply core/google/api/core/helpers/grpc_helpers.py]: Done.

**Pull request merged**